api/internal/store/service/live.go

```diff
@@ -64,3 +65,33 @@ func (s *service) GetLive(ctx context.Context, in *store.GetLiveInput) (*entity.
 	live.FillIVS(*fillIvsParams)
 	return live, nil
 }
+
+func (s *service) UpdateLivePublic(ctx context.Context, in *store.UpdateLivePublicInput) error {
+	if err := s.validator.Struct(in); err != nil {
+		return exception.InternalError(err)
+	}
+
+	_, err := s.db.Live.Get(ctx, in.LiveID)
+	if err != nil {
+		return exception.InternalError(err)
+	}
+
+	ivsParams := &ivs.CreateChannelParams{
+		LatencyMode: "NORMAL",
+		Name:        in.ChannelName,
+		ChannelType: "BASIC",
+	}
+	cout, err := s.ivs.CreateChannel(ctx, ivsParams)
+	if err != nil {
+		return exception.InternalError(err)
+	}
+	dbParams := &database.UpdateLivePublicParams{
+		Published:    in.Published,
+		Canceled:     in.Canceled,
+		ChannelArn:   aws.ToString(cout.Channel.Arn),
+		StreamKeyArn: aws.ToString(cout.StreamKey.Arn),
+	}
+
+	err = s.db.Live.UpdatePublic(ctx, in.LiveID, dbParams)
+	return exception.InternalError(err)
+}
```

Comment:
このメソッド、空行の使い方が気持ち悪いかも..👀

空行入れるなら、↓の感じで処理ごとに分けるほうがいいかも

```suggestion
func (s *service) UpdateLivePublic(ctx context.Context, in *store.UpdateLivePublicInput) error {
	if err := s.validator.Struct(in); err != nil {
		return exception.InternalError(err)
	}

	_, err := s.db.Live.Get(ctx, in.LiveID)
	if err != nil {
		return exception.InternalError(err)
	}

	ivsParams := &ivs.CreateChannelParams{
		LatencyMode: "NORMAL",
		Name:        in.ChannelName,
		ChannelType: "BASIC",
	}
	cout, err := s.ivs.CreateChannel(ctx, ivsParams)
	if err != nil {
		return exception.InternalError(err)
	}

	dbParams := &database.UpdateLivePublicParams{
		Published:    in.Published,
		Canceled:     in.Canceled,
		ChannelArn:   aws.ToString(cout.Channel.Arn),
		StreamKeyArn: aws.ToString(cout.StreamKey.Arn),
	}
	err = s.db.Live.UpdatePublic(ctx, in.LiveID, dbParams)
	return exception.InternalError(err)
}
```